src/robotide/controller/commands.py

```diff
@@ -279,12 +280,35 @@ class NullObserver(object):
 class RenameKeywordOccurrences(_ReversibleCommand):
 
     def __init__(self, original_name, new_name, observer, keyword_info=None):
-        self._original_name = original_name
-        self._new_name = new_name
+        keyword_name, new_keyword_name = self._check_gherkin(new_name,
+                                                             original_name)
+        self._original_name = keyword_name
+        self._new_name = new_keyword_name
         self._observer = observer
         self._keyword_info = keyword_info
         self._occurrences = None
 
+    def _check_gherkin(self, new_name, original_name):
+        _gherkin_prefix = re.compile('^(Given|When|Then|And|But) ',
+                                     re.IGNORECASE)
+        keyword_name = re.sub(_gherkin_prefix, '', original_name)
+        was_gherkin = (keyword_name != original_name)
```

Comment:
I think these two lines should be extracted as a helper method, since they are repeated below.